File: jitlibrary-x86_64/src/main/c/ITLibArtist.m
/*
 * =================================================
 * Copyright 2006-2019 tagtraum industries incorporated
 * All rights reserved.
 * =================================================
 *
 * @author <a href="mailto:hs@tagtraum.com">Hendrik Schreiber</a>
 */
#import "com_tagtraum_jitlibrary_ITLibArtist.h"
#import "Util.h"
#import <iTunesLibrary/ITLibArtist.h>


JNIEXPORT jlong JNICALL Java_com_tagtraum_jitlibrary_ITLibArtist_getId
        (JNIEnv *env, jobject instance) {
    ENTER
    ITLibMediaEntity *artist = (ITLibMediaEntity *) getPointer(env, instance);
    if (artist == NULL) {
        return 0L;
    } else {
        return getPersistentId(artist);
    }
    EXIT(env)
    return 0L;
}

/*
 * Class:     com_tagtraum_jitlibrary_ITLibArtist
 * Method:    getName
 */
JNIEXPORT jstring JNICALL Java_com_tagtraum_jitlibrary_ITLibArtist_getName
        (JNIEnv *env, jobject instance) {
    ENTER
    ITLibArtist *artist = (ITLibArtist *) getPointer(env, instance);
    if (artist == NULL) {
        return NULL;
    } else {
        return createJavaStringFromNSString(env, artist.name);
    }
    EXIT(env)
    return NULL;
}

/*
 * Class:     com_tagtraum_jitlibrary_ITLibArtist
 * Method:    getSortName
 * Signature: ()Ljava/lang/String;
 */
JNIEXPORT jstring JNICALL Java_com_tagtraum_jitlibrary_ITLibArtist_getSortName
        (JNIEnv *env, jobject instance) {
    ENTER
    ITLibArtist *artist = (ITLibArtist *) getPointer(env, instance);
    if (artist == NULL) {
        return NULL;
    } else {
        return createJavaStringFromNSString(env, artist.sortName);
    }
    EXIT(env)
    return NULL;
}

File: jitlibrary-java/src/main/java/com/tagtraum/jitlibrary/ITLibArtist.java
/*
 * =================================================
 * Copyright 2019 tagtraum industries incorporated
 * All rights reserved.
 * =================================================
 */
package com.tagtraum.jitlibrary;

/**
 * Represents an artist, such as the performer of a song.
 *
 * @author <a href="mailto:hs@tagtraum.com">Hendrik Schreiber</a>
 * @see <a href="https://developer.apple.com/documentation/ituneslibrary/itlibartist">Apple ITLibArtist</a>
 */
public class ITLibArtist extends ReferenceCountedObject {

    public ITLibArtist(final long pointer) {
        super(pointer);
    }

    public native long getId();

    public native String getName();

    public native String getSortName();

}
